datafusion/physical-plan/src/aggregates/row_hash.rs

```diff
@@ -194,7 +194,9 @@ impl SkipAggregationProbe {
         if self.input_rows >= self.probe_rows_threshold {
             self.should_skip = self.num_groups as f64 / self.input_rows as f64
                 >= self.probe_ratio_threshold;
-            self.is_locked = true;
+            // Set is_locked to true only if we have decided to skip, otherwise we can try to skip
+            // during processing the next record_batch.
+            self.is_locked = self.should_skip;
```

Comment:
Another way to write this that might make it clearer what the valid states are could be

```rust
            if self.num_groups as f64 / self.input_rows as f64 >= self.probe_ratio_threshold {
              self.should_skip = true;
              self.is_locked = true;
            }
```

We could also potentially make it a tri state enum rather than two flags
```rust 
enum ProbeState {
  // actively probing, not skipping, checking threshold
  Active,
  // Locked, not skipping, no longer checking threshold
  Locked, 
  // Skipping initial phase, no longer checking threshold
  Skipping
}
```